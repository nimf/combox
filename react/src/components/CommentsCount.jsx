import React from 'react';
import PropTypes from 'prop-types';
import { Header } from 'semantic-ui-react';

const propTypes = {
  count: PropTypes.number,
};

const defaultProps = {
  count: undefined,
};

function CommentsCount({ count }) {
  let displayCount = 'Loading comments...';
  if (count !== undefined) {
    displayCount = count > 0
      ? `${count} comment${count > 1 ? 's' : ''}`
      : 'No comments yet';
  }
  return <Header as="h4">{displayCount}</Header>;
}

CommentsCount.propTypes = propTypes;
CommentsCount.defaultProps = defaultProps;

export default CommentsCount;
