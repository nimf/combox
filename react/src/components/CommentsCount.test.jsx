import React from 'react';
import renderer from 'react-test-renderer';
import CommentsCount from './CommentsCount';

describe('CommentsCount renders', () => {
  test('CommentsCount renders undefined count', () => {
    const tree = renderer.create(<CommentsCount
      count={undefined}
    />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('CommentsCount renders zero count', () => {
    const tree = renderer.create(<CommentsCount
      count={0}
    />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('CommentsCount renders 1 count', () => {
    const tree = renderer.create(<CommentsCount
      count={1}
    />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('CommentsCount renders 2 count', () => {
    const tree = renderer.create(<CommentsCount
      count={2}
    />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
