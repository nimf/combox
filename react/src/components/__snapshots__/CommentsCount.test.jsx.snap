// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentsCount renders CommentsCount renders 1 count 1`] = `
<h4
  className="ui header"
>
  1 comment
</h4>
`;

exports[`CommentsCount renders CommentsCount renders 2 count 1`] = `
<h4
  className="ui header"
>
  2 comments
</h4>
`;

exports[`CommentsCount renders CommentsCount renders undefined count 1`] = `
<h4
  className="ui header"
>
  Loading comments...
</h4>
`;

exports[`CommentsCount renders CommentsCount renders zero count 1`] = `
<h4
  className="ui header"
>
  No comments yet
</h4>
`;
